packages/dashboard-core-plugins/src/panels/ChartPanel.tsx

```diff
@@ -122,9 +122,18 @@ export interface ChartPanelProps extends DashboardPanelProps {
   metadata: ChartPanelMetadata;
   /** Function to build the ChartModel used by this ChartPanel. Can return a promise. */
   makeModel: () => Promise<ChartModel>;
+  localDashboardId: string;
+  Plotly?: typeof PlotlyType;
+  /** The panel container div */
+  containerRef?: RefObject<HTMLDivElement>;
+
+  panelState: GLChartPanelState;
+  settings: Partial<WorkspaceSettings>;
+}
+
+interface PropsFromRedux {
```

Comment:
We've done this paradigm before, but slightly differently. Take a look at `Linker.tsx`, where we define `StateProps`, `OwnProps`, and then the `connector` and join them together. I think overall that way is a little bit cleaner - you can actually use `OwnProps` in `mapStateToProps`, and you don't need to redefine all the types from redux, they will implicitly get picked up from the connector.

Reply:
You still need to do `OwnProps & StateProps` so the component knows, which is just a different name for what I did here. I changed to `OwnProps` and `StateProps` though.

A downside is if you need to use `OwnProps` from another file, it should be more specific. Minor thing, but `PandasPanel` needs to use `IrisGridOwnProps` because it doesn't not get/set the props from redux itself, but renders the connected component which does it. If you just use the `IrisGridProps` as `PandasProps` type, TS expects you to pass the redux props as well since it's not a connected component

Reply:
Within `IrisGridPanel.tsx`, just name it `OwnProps`; then the importer can rename it as needed, e.g. `import { type OwnProps as IrisGridOwnProps } from './IrisGridPanel'`. It's already namespaced within the file, the extra namespacing is a little redundant (though we do this in a ton of places).
The main advantage is not needing to specify `DispatchProps`, but you do need to move the `connect` and `mapStateToProps` up.